sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/UploadBufferPool.java

```diff
@@ -55,26 +39,22 @@
      * Creates a new instance of UploadBufferPool
      * @param numBuffs The number of buffers in the buffer pool.
      * @param buffSize The size of the buffers
+     * @deprecated Use {@link UploadBufferPool#UploadBufferPool(long, long)}
      */
+    @Deprecated
     public UploadBufferPool(final int numBuffs, final long buffSize, long maxBuffSize) {
```

Comment:
can I just remove this since its in an implementation package?

Reply:
Let's keep it for a while for people who might get versions wrong, so they don't see class not found exceptions.

Reply:
on the other hand. if they pick wrong implementation they'd suffer from hangs... maybe better to fail fast and prompt that they got dependencies wrong. Let's remove it.